src/fsharp/TcGlobals.fs

```diff
@@ -360,7 +361,7 @@ type public TcGlobals(compilingFslib: bool, ilg:ILGlobals, fslibCcu: CcuThunk, d
   let mkForallTyIfNeeded d r = match d with [] -> r | tps -> TType_forall(tps, r)
 
       // A table of all intrinsics that the compiler cares about
-  let v_knownIntrinsics = Dictionary<(string * string option * string * int), ValRef>(HashIdentity.Structural)
+  let v_knownIntrinsics = ConcurrentDictionary<(string * string option * string * int), ValRef>(HashIdentity.Structural)
```

Comment:
We can use struct tuple keys here

Reply:
I'm not particularly keen on that.  Struct keys will get copied a lot anyway, and this is not a hot path for anything